lms/static/scripts/frontend_apps/services/vitalsource.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * @typedef {import('../api-types').Book} Book
+ * @typedef {import('../api-types').Chapter} Chapter
+ */
+
+import { ApiError } from '../utils/api';
+import { bookList, chapterData } from '../utils/vitalsource-sample-data';
+
+/** @param {number} ms */
+function delay(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+/**
+ * Service for fetching information about available VitalSoure books and the
+ * list of chapters in a book.
+ */
+export class VitalSourceService {
+  /**
+   * @param {object} options
+   *   @param {string} options.authToken
+   */
+  constructor({ authToken }) {
+    // This field is not currently used. It will be in future to fetch book
+    // and chapter info from the backend.
+    this._authToken = authToken;
```

Comment:
Is there any foreseeable drawback with multiple services maintaining their own auth-token attributes?

Reply:
As long as the auth token doesn't change during a session this is fine. If in future we had a situation where the auth token might change during a session then we'd need some centralized storage of the token.